bin/test-scratch

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+set -eu
+
+while :; do
+  while getopts ":a:e:p:s:" arg; do
+    case $arg in
+      a)
+        auth_file_key="$OPTARG"
+        ;;
+      e) 
+        enc_auth_file="$OPTARG"
+        ;;
+      p)
+        source_path="$OPTARG"
+        ;;
+      s)
+        scratch_def_file="$OPTARG"
+        ;;
+      *)
+        echo Ignoring unknown option: $OPTARG
+        ;;
+    esac
+  done
+  OPTIND=$((OPTIND+1))
+  [ $OPTIND -gt $# ] && break
+done
+
+# Decrypt Auth URL File
+openssl enc -d -aes-256-cbc -md md5 -in ${enc_auth_file} -out auth_url.txt -k ${auth_file_key}
+
+# Authenticate DevHub Org
+sfdx force:auth:sfdxurl:store -f auth_url.txt -d -a DevHub && rm auth_url.txt
+
+# Create Scratch Org
+sfdx force:org:create -v DevHub -f ${scratch_def_file} -s -a TargetOrg -d 1
+
+# Push Source
+sfdx force:source:deploy -p ${source_path} -u TargetOrg
+
+# Run Test(s)
+sfdx force:apex:test:run -c -r human -u TargetOrg
+
+# Mark Org for Deletion
+sfdx force:org:delete -p -u TargetOrg
```

Comment:
I see `test-scratch` being more useful for 2GP projects as opposed to single-org projects where the `validate` command can be used to run tests in production directly